### PR TITLE
[UI] Add SynForm field wrappers for selects, switches, and custom controls (AAP-92071)

### PR DESCRIPTION
**Pull request by skochay330**
## Summary
- Add five SynForm field wrappers (`SynSelectField`, `SynMultiSelectField`, `SynSwitchField`, `SynFileField`, `SynExpressionField`) built on the existing `SynFormField` render-prop pattern for upcoming form migrations.
- Extend `SynFormField` with `hideFormGroupLabel` and `hideFooter` for switch and expression field layouts.
- Consolidate shared expression input logic into `ExpressionFieldInput` (used by `SynExpressionField` and builder `ExpressionFormField`) and move builder drag constants to `components/expressions/`.

## Out of scope
- No call-site migrations yet (`ImportWorkflowDialog`, identity provider switches, group multi-select, etc.) — wrappers are primitives only.
- Storybook screenshots not attached in this PR.

## Test plan
- [x] `npx vitest run` on new/changed form wrapper tests (150 tests)
- [x] ESLint clean on modified frontend files
- [x] TypeScript clean (`npx tsc`)
- [x] Existing `ExpressionFormField` tests pass after shared input refactor
- [ ] Reviewer: spot-check Storybook stories for the five new wrappers

Fixes AAP-92071

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: frontend/packages/syntara-ui/src/components/forms/SynFileField.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        const handleFileInputChange = (_event: DropEvent, inputFile: File) => {
+          field.onChange(inputFile)
+        }
+
+        const handleClearClick = () => {
+          field.onChange(undefined)
+        }
+
+        return (
+          <FileUpload
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/components/forms/SynFileField.tsx]: other wrappers wire `field.onBlur` and `field.name`. probably fine for submit-time validation, but worth matching if a consumer uses `mode: 'onBlur'`.

File: frontend/packages/syntara-ui/src/components/forms/SynSelectField.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+  const renderToggle = useCallback(
+    (toggleRef: Ref<MenuToggleElement>) => renderSynSelectFieldToggle(toggleRef, toggleState),
+    [toggleState]
+  )
+
+  return (
+    <SynSelect
+      id={resolvedFieldId}
+      aria-label={label}
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/components/forms/SynSelectField.tsx]: toggle accessible name comes from placeholder/selection, not the visible FormGroup label. `aria-label={label}` may be redundant here, or not what we want if PF already links via `fieldId`. worth a quick screen reader pass.

File: frontend/packages/syntara-ui/src/components/forms/SynSwitchField.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  const resolvedFieldId = fieldId ?? name
+
+  return (
+    <SynFormField
+      name={name}
+      control={control}
+      label={label}
+      fieldId={resolvedFieldId}
+      hideFormGroupLabel
+      labelHelp={labelHelp}
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/components/forms/SynSwitchField.tsx]: `hideFormGroupLabel` hides the group label but we still pass `labelHelp`. untested edge case. help icon placement might be odd without a visible group label.

**Review by gitdallas — COMMENTED** [review]
no blockers. primitives look good, tests/axe coverage is solid, and the ExpressionFormField extraction is clean. nits inline.

File: frontend/packages/syntara-ui/src/routes/builder/panels/fields/ExpressionFormField.tsx [hunk 4]
```diff
@@ -101,22 +32,14 @@ function ExpressionFormField({
       fieldId={id}
       data-drop-target={isDropTarget ? 'active' : 'inactive'}
     >
-      <TextInput
+      <ExpressionFieldInput
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/routes/builder/panels/fields/ExpressionFormField.tsx:37]: builder errors now go through `FormFieldError` (with icon) instead of plain `HelperTextItem`. fine if intentional alignment with SynForm fields, just flagging the visible builder tweak.

File: frontend/packages/syntara-ui/src/components/forms/SynMultiSelectField.tsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+  field: ControllerRenderProps<TFieldValues, TName>
+  fieldState: ControllerFieldState
+  resolvedFieldId: string
+  label: string
+  options: SynSelectFieldOption[]
+  placeholder: string
+  isDisabled?: boolean
+}
+
+function SynMultiSelectFieldControl<TFieldValues extends FieldValues, TName extends FieldPath<TFieldValues>>({
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/components/forms/SynMultiSelectField.tsx:35]: SynSelectField control shell is nearly identical. fine for primitives-only, but might be worth extracting before the migration PRs land.

File: frontend/packages/syntara-ui/src/components/expressions/expressionFieldDrag.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  contextPath: string
+}
+
+export type DragData = FieldDragData | ContextDragData
+
+export function isDragData(value: unknown): value is DragData {
+  if (typeof value !== 'object' || value === null) return false
+  const obj = value as Record<string, unknown>
+  if (obj.type === DRAG_TYPE_FIELD) {
+    return typeof obj.nodeId === 'string' && Array.isArray(obj.fieldPath)
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/src/components/expressions/expressionFieldDrag.ts]: security bot flagged this. false positive IMO. drag JSON becomes expression text and React escapes on render. optional stricter shape checks if we want defense in depth.

File: frontend/packages/syntara-ui/src/components/forms/SynFileField.tsx (hunk 1 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/components/forms/SynFileField.tsx]: Good catch — wired `field.onBlur` via a focus-leaving wrapper around `FileUpload` (PF does not expose blur on the control). PatternFly also does not let us set `name` on the hidden file input, so that prop remains unwired here.

File: frontend/packages/syntara-ui/src/components/forms/SynSelectField.tsx (hunk 2 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/components/forms/SynSelectField.tsx]: Removed the redundant `aria-label` on `SynSelect`. The visible `FormGroup` label is associated via `fieldId`, and the toggle text reflects the placeholder/selection. Added a test asserting the label `for` attribute matches the field id.

File: frontend/packages/syntara-ui/src/components/forms/SynSwitchField.tsx (hunk 3 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/components/forms/SynSwitchField.tsx]: Agreed — with `hideFormGroupLabel`, `labelHelp` placement is awkward. Dropped `labelHelp` from `SynSwitchField` and documented using `hint` for inline help instead.

File: frontend/packages/syntara-ui/src/routes/builder/panels/fields/ExpressionFormField.tsx (hunk 4 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/routes/builder/panels/fields/ExpressionFormField.tsx:37]: Intentional — `ExpressionFieldInput` uses `FormFieldError` so builder expression fields match SynForm validation styling. Added an inline comment in `ExpressionFormField` to make that explicit.

File: frontend/packages/syntara-ui/src/components/forms/SynMultiSelectField.tsx (hunk 5 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/components/forms/SynMultiSelectField.tsx:35]: Extracted the shared toggle/menu shell into `synSelectFieldControlShell.tsx`; single- and multi-select now only differ in selection logic and option rendering.

File: frontend/packages/syntara-ui/src/components/expressions/expressionFieldDrag.ts (hunk 6 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/components/expressions/expressionFieldDrag.ts]: Added stricter shape checks in `isDragData` (non-empty strings, non-empty `fieldPath` segments) for defense in depth. Agree the security finding is a false positive — dragged JSON becomes expression text and React escapes on render.

**Review by gitdallas — CHANGES_REQUESTED** [review]
follow-up commit addressed the first-round nits (file blur, select shell extract, switch labelHelp, drag validation). one blocker on the new label test.

`SynSelectField.test.tsx` asserts `for` on the text node from `getByText('Project')`. PF6 puts label text in a child span, so it fails locally (152/153 form tests pass otherwise). use `container.querySelector('label[for=\"project-id\"]')` or `getByText('Project').closest('label')`.

also drop the `eslint-disable` on the `SynFileField` blur wrapper. lint passes without it.

rest looks good. approve once CI is green on those.

**Review by gitdallas — CHANGES_REQUESTED** [review]
prior feedback mostly landed

still blocked on `(Frontend) Checks`:
- `SynSelectField.test.tsx:28` `container.querySelector(...)` hits `testing-library/no-node-access`. lines 29-30 already assert label association via `getByLabelText('Project')`; drop the querySelector line.
- run prettier on `SynFileField.tsx` (format:check failed in CI).

approve once checks pass.

**Review by gitdallas — APPROVED** [review]

**Review by mfrances17 — APPROVED** [review]
👍 LGTM
